mountpoint-s3-fs/src/superblock.rs

```diff
@@ -98,6 +102,11 @@ impl Superblock {
             config.cache_config.negative_cache_ttl,
         );
 
+        let manifest = config
+            .manifest_db_path
+            .as_ref()
+            .map(|manifest_db_path| Manifest::new(manifest_db_path).expect("manifest must be created"));
```

Comment:
We should not panic here. Have you considered adding the pre-built `Manifest` to `config`, instead of the path? That would shift the burden of initializing it to the caller (only when they use the manifest).

The alternative is to return an error here, but it would be cumbersome to change the signature only for this edge case.